el-patch-template.el

```diff
@@ -0,0 +1,554 @@
+                             (puthash template old-entry table)
+                             ;; and rethrow
+                             (throw 'no-match nil)))))))
+            (el-patch-template--process form
+                                        (or
+                                         ;; The previous resolution
+                                         (cdr symbol)
+                                         ;; The template-value
+                                         (car symbol))
+                                        nil
+                                        symbol-next-step
+                                        table literal))
+          (throw 'no-match nil))))))
+
+(defun el-patch-template--match-p (form template)
+  "Check if the forms in FORM match TEMPLATE.
+TEMPLATE may contain `...' which greedily matches any number of
+forms in FORM. Match is successful if a partial list of FORM,
+starting from the beginning, matches TEMPLATE. The return value
+is the number of forms in FORM which match TEMPLATE or nil if a
+match is not possible."
+  (cond
+   ((and (consp template) (consp form))
+    (when-let ((matched-count
+                (if (member (car template) '(...))
+                    (or
+                     (el-patch-template--match-p (cdr form)
+                                                 template)
+                     ;; If we are here, we failed so try consuming
+                     ;; `...'
+                     (el-patch-template--match-p (cdr form)
+                                                 (cdr template)))
+                  (and
+                   (el-patch-template--match-p (car form)
+                                               (car template))
+                   (el-patch-template--match-p (cdr form)
+                                               (cdr template))))))
+      (1+ matched-count)))
+   ((and (vectorp template) (vectorp form))
+    (el-patch-template--match-p (append form nil);; covert to list
+                                (append template nil)))
+   ((and (consp template)
+         (equal (car template) 'el-patch-template--concat)
+         (stringp form))
+    (string-match-p
+     (apply 'concat (mapcar (lambda (x)
+                              (if (and (equal x '...))
+                                  ;; match any character
+                                  ;;"[\0-\377[:nonascii:]]*"
+                                  "\\(.\\|\n\\)*"
+                                (regexp-quote x)))
+                            (cdr template)))
+     form))
+   (t (or (and (null template) 0)
+          (and (or (member template '(...))
+                   (equal template form))
+               (if (consp form)
+                   (length form)
+                 1))))))
+
+(defun el-patch-template--any-p (definition ptemplates &optional up-to)
+  "Return t if any template in PTEMPLATES matches any form in DEFINITION.
```

Comment:
`checkdoc` complains about "matches" here, but it refers to the argument rather than the function!

Reply:
Yeah, this is silly. You can fix it by adding

```elisp
;; checkdoc-verb-check-experimental-flag: nil
```

to the `Local Variables` list at the bottom of the file.